src/languages/javascript.js

```diff
@@ -148,13 +149,30 @@ export default function(hljs) {
       hljs.C_BLOCK_COMMENT_MODE,
       NUMBER,
       { // object attr container
-        begin: /[{,\n]\s*/, relevance: 0,
+        begin: regex.concat(/[{,\n]\s*/,
+          // we need to look ahead to make sure that we actually have an
+          // attribute coming up so we don't steal a comma from a potential
+          // "value" container
+          //
+          // NOTE: this might not work how you think.  We don't actually always
+          // enter this mode and stay.  Instead it might merely match `,
+          // <comments up next>` and then immediately end after the , because it
+          // fails to find any actual attrs. But this still does the job because
+          // it prevents the value contain rule from grabbing this instead and
+          // prevening this rule from firing when we actually DO have keys.
+          regex.lookahead(regex.concat(
+            // we also need to allow for multiple possible comments inbetween
+            // the first key:value pairing
+            /(\/\/.*\s*)*/,
+            /(\/\*.*\*\/\s*)*/,
+            IDENT_RE + '\\s*:'))),
```

Comment:
This trick doesn't work in case like this:
```
  key5: value, /* with a multiline comment */ // another comment
  key6: value,
```
or like this:
```
  key5: value, /* with a REAL multiline
comment */
  key6: value,
```

Reply:
Good catch, pretty sure I can fix those cases.

Reply:
Do you understand why the last one doesn't work?

Reply:
Better?

Reply:
Yes, it seems fine.
(Still it looks somewhat ugly, although I can't see how to handle it more elegantly. We can't express it by saying "first match all comments and then just ignore them altogether")

Reply:
Our syntax isn't expressive enough. :-)  You could have something like:

```js
alwaysIgnore: COMMENTS,
match: ["{", followedBy: "[ident]: "]
```

Where match is pretty much any sequence and then the always ignore item would turn into a huge regex inserted in the middle that was always ignored. Although I guess here I'm talking about look-ahead where it's easy... as soon as you want to emit them while you're ignoring them then it's a whole other thing. :-)

Although I'm not sure that'd be much different actually... just a way of inserting random modes in-between any other modes... seems most useful for comments... but for another day...